BuildAssets/repos/test-rpm.sh

```diff
@@ -1,21 +1,46 @@
 #!/bin/bash
 
-# This script requires environment variables to run:
-#   S3_PUBLISH_ENDPOINT, OCTOPUS_CLI_SERVER, OCTOPUS_CLI_API_KEY, OCTOPUS_SPACE, OCTOPUS_EXPECT_ENV
+if [[ -z "$PUBLISH_LINUX_EXTERNAL" || -z "$OCTOPUS_CLI_SERVER" || -z "$OCTOPUS_CLI_API_KEY" \
+   || -z "$OCTOPUS_SPACE" || -z "$OCTOPUS_EXPECT_ENV" ]]; then
+  echo -e 'This script requires the following environment variables to be set:
+  PUBLISH_LINUX_EXTERNAL, OCTOPUS_CLI_SERVER, OCTOPUS_CLI_API_KEY, OCTOPUS_SPACE, OCTOPUS_EXPECT_ENV' >&2
+  exit 1
+fi
 
-# Configure yum
-curl -s "http://$S3_PUBLISH_ENDPOINT/tentacle.repo" -o /etc/yum.repos.d/tentacle.repo || exit
-curl -s "http://$S3_PUBLISH_ENDPOINT/octopuscli.repo" -o /etc/yum.repos.d/octopuscli.repo || exit
+if [[ "$PUBLISH_LINUX_EXTERNAL" == "true" ]]; then
+  ORIGIN="https://rpm.octopus.com"
+else
+  ORIGIN="http://prerelease.rpm.octopus.com"
+fi
 
-# Install
-yum --quiet --assumeyes install tentacle 2>&1 || exit
-yum --quiet --assumeyes install octopuscli 2>&1 || exit
+# The following should be kept in sync with our documented download instructions (but more automated and quieter).
 
-# Test
-echo "== Testing Tentacle =="
+echo "## Configuring yum"
+curl --silent --show-error --fail --location "$ORIGIN/tentacle.repo" --output /etc/yum.repos.d/tentacle.repo || exit
+curl --silent --show-error --fail --location "$ORIGIN/octopuscli.repo" --output /etc/yum.repos.d/octopuscli.repo || exit
+
+echo "## Installing tentacle, octopuscli"
+if [[ $(. /etc/os-release && echo $ID) != "rhel" ]]; then
+  yum --quiet --assumeyes install tentacle octopuscli 2>&1 || exit
+else
+  if [[ -z "$REDHAT_SUBSCRIPTION_USERNAME" || -z "$REDHAT_SUBSCRIPTION_PASSWORD" ]]; then
+    echo -e 'Unable to test in RHEL without environment variables: REDHAT_SUBSCRIPTION_USERNAME, REDHAT_SUBSCRIPTION_PASSWORD.' >&2
+    exit 1
+  fi
+  subscription-manager register --username "$REDHAT_SUBSCRIPTION_USERNAME" \
+    --password "$REDHAT_SUBSCRIPTION_PASSWORD" --auto-attach >/dev/null 2>&1 || { echo "Red Hat subscribe problem." >&2; exit 1; }
```

Comment:
This might suppress errors a bit too much

Reply:
I've adjusted these to capture output, but only display it on error, to keep spam under control during successful runs.